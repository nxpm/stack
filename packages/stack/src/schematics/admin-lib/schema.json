{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "AdminLib",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the library?",
      "pattern": "^[a-zA-Z]{1}.*$"
    },
    "type": {
      "description": "The type of library",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": {
        "message": "Which type of library would you like to create?",
        "type": "list",
        "items": [
          {
            "value": "assets",
            "label": "Assets library"
          },
          {
            "value": "data-access",
            "label": "Data access library"
          },
          {
            "value": "feature",
            "label": "Feature library"
          },
          {
            "value": "style",
            "label": "Style Library"
          },
          {
            "value": "ui",
            "label": "UI Library"
          },
          {
            "value": "util",
            "label": "Utility library"
          }
        ]
      }
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "publishable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a publishable library."
    },
    "buildable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a buildable library."
    },
    "prefix": {
      "type": "string",
      "format": "html-selector",
      "description": "The prefix to apply to generated selectors.",
      "alias": "p"
    }
  },
  "required": ["name"]
}
