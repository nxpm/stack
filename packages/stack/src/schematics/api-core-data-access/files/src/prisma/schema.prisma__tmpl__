datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}

model User {
  id                   String           @id @default(cuid())
  createdAt            DateTime         @default(now())
  updatedAt            DateTime         @default(now()) @updatedAt
  role                 Role
  developer            Boolean          @default(false)
  email                String           @unique
  username             String           @unique
  password             String?
  firstName            String?
  lastName             String?
  avatarUrl            String?
  location             String?
  phone                String?
  bio                  String?
}

enum Role {
  Admin
  User
}

