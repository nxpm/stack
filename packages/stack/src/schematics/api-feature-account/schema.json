{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "ApiFeatureUser",
  "title": "",
  "type": "object",
  "properties": {
    "appName": {
      "type": "string",
      "description": "Name of the app to connect to",
      "alias": "a"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    }
  },
  "required": []
}
