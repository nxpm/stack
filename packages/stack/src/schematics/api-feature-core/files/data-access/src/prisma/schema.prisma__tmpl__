datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}

model User {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  updatedAt DateTime @default(now()) @updatedAt
  role      Role
  developer Boolean  @default(false)
  username  String   @unique
  password  String?
  firstName String?
  lastName  String?
  avatarUrl String?
  location  String?
  phone     String?
  bio       String?
  emails    Email[]  @relation("Email_belongsTo_Owner")
}

enum Role {
  Admin
  User
}

model Email {
  id            String    @id @default(cuid())
  createdAt     DateTime  @default(now())
  updatedAt     DateTime  @default(now()) @updatedAt
  email         String    @unique
  public        Boolean   @default(false)
  primary       Boolean   @default(false)
  verified      Boolean   @default(false)
  verifyToken   String?
  verifyExpires DateTime?
  owner         User      @relation(name: "Email_belongsTo_Owner", fields: [ownerId], references: [id])
  ownerId       String
  // Enable when this is resolved https://github.com/prisma/prisma/issues/3387
  // @@unique([ownerId, primary])
}
