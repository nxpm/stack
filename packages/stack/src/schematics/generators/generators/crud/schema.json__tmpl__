{
  "cli": "nx",
  "id": "web-module",
  "type": "object",
  "properties": {
    "target": {
      "type": "string",
      "description": "Target folder to create the module",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "name": {
      "type": "string",
      "description": "Name of the module (in kebab-case)",
      "$default": {
        "$source": "argv",
        "index": 1
      }
    }
  },
  "required": ["target", "name"]
}
