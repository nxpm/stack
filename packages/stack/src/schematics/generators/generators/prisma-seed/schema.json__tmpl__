{
  "cli": "nx",
  "id": "prisma-seed",
  "type": "object",
  "properties": {
    "confirm": {
      "type": "boolean",
      "description": "Library confirm",
      "x-prompt": "Are you sure you want to delete and seed the database?"
    },
    "timeout": {
      "type": "number",
      "description": "Number of seconds to wait before before starting deletion",
      "default": 10
    }
  },
  "required": ["confirm", "timeout"]
}
