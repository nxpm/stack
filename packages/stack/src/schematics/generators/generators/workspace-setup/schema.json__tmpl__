{
  "cli": "nx",
  "id": "workspace-setup",
  "type": "object",
  "properties": {},
  "required": []
}
