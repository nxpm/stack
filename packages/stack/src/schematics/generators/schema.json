{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "Init",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "default": "web"
    }
  },
  "required": []
}
