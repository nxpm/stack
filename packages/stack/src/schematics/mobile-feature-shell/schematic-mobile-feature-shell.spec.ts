import { Tree } from '@angular-devkit/schematics'
import { SchematicTestRunner } from '@angular-devkit/schematics/testing'
import { createEmptyWorkspace } from '@nrwl/workspace/testing'
import { join } from 'path'

import { MobileFeatureShellSchematicSchema } from './schema'

describe('mobile-feature-shell schematic', () => {
  let appTree: Tree
  const options: MobileFeatureShellSchematicSchema = { appName: 'test' }

  const testRunner = new SchematicTestRunner('@nxpm/mobile-feature-shell', join(__dirname, '../../../collection.json'))

  beforeEach(() => {
    appTree = createEmptyWorkspace(Tree.empty())
  })

  it('should run successfully', async () => {
    await expect(
      testRunner.runSchematicAsync('mobile-feature-shell', options, appTree).toPromise(),
    ).resolves.not.toThrowError()

    expect(appTree.read('libs/test/shell/feature/src/lib/test-shell-feature.module.ts').toString()).toMatchSnapshot()
  })
})
