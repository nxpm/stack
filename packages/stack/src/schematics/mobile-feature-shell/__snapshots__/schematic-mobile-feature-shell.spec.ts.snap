// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mobile-feature-shell schematic should run successfully 1`] = `
"import { NgModule } from '@angular/core'
import { PreloadAllModules, RouteReuseStrategy, RouterModule, Routes } from '@angular/router'
import { SplashScreen } from '@ionic-native/splash-screen/ngx'
import { StatusBar } from '@ionic-native/status-bar/ngx'
import { IonicModule, IonicRouteStrategy } from '@ionic/angular'
import { TestAuthDataAccessModule, IsLoggedInGuard } from '@nxpm/test/auth/data-access'
import { TestCoreFeatureModule } from '@nxpm/test/core/feature'
import { TestLayoutFeatureComponent } from '@nxpm/test/layout/feature'

const routes: Routes = [
  {
    path: '',
    component: TestLayoutFeatureComponent,
    canActivate: [IsLoggedInGuard],
    children: [
      // Application routes here
      { path: '', pathMatch: 'full', redirectTo: 'dashboard' },
      {
        path: 'about',
        loadChildren: () => import('@nxpm/test/about/feature').then((m) => m.TestAboutFeatureModule),
      },
      {
        path: 'account',
        loadChildren: () => import('@nxpm/test/account/feature').then((m) => m.TestAccountFeatureModule),
      },
      {
        path: 'dashboard',
        loadChildren: () => import('@nxpm/test/dashboard/feature').then((m) => m.TestDashboardFeatureModule),
      },
      { path: 'not-found', loadChildren: () => import('./not-found/not-found.module').then((m) => m.NotFoundModule) },
    ],
  },
  {
    path: '',
    loadChildren: () => import('@nxpm/test/auth/feature').then((m) => m.TestAuthFeatureModule),
  },
  { path: '**', redirectTo: '/not-found' },
]

@NgModule({
  providers: [StatusBar, SplashScreen, { provide: RouteReuseStrategy, useClass: IonicRouteStrategy }],
  imports: [
    IonicModule.forRoot(),
    RouterModule.forRoot(routes, { paramsInheritanceStrategy: 'always', preloadingStrategy: PreloadAllModules }),
    TestAuthDataAccessModule,
    TestCoreFeatureModule,
  ],
})
export class TestShellFeatureModule {}
"
`;
